packages/lesswrong/components/posts/FeedPostsHighlight.tsx

```diff
@@ -0,0 +1,217 @@
+import { Components, registerComponent } from '../../lib/vulcan-lib';
+import { postGetPageUrl } from '../../lib/collections/posts/helpers';
+import React, { FC, useState, useCallback, useEffect } from 'react';
+import { Link } from '../../lib/reactRouterWrapper';
+import { useSingle } from '../../lib/crud/withSingle';
+import { nofollowKarmaThreshold } from '../../lib/publicSettings';
+import { useForeignCrosspost, isPostWithForeignId, PostWithForeignId } from "../hooks/useForeignCrosspost";
+import { useForeignApolloClient } from "../hooks/useForeignApolloClient";
+import { captureException }from "@sentry/core";
+import classNames from 'classnames';
+import { truncateWithGrace } from '../common/ContentItemTruncated';
+import { useRecordPostView } from '../hooks/useRecordPostView';
+import { useTracking } from '../../lib/analyticsEvents';
+
+const styles = (theme: ThemeType) => ({
+  root: {
+
+    '& .read-more-button': {
+      // color: theme.palette.primary.main,
+      opacity: 0.7,
+    }
+  },
+  expandedTextBody: {
+    marginBottom: 10,
+  },
+  highlightContinue: {
+    marginTop:theme.spacing.unit*2,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    '&& a, && a:hover': {
+      color: theme.palette.primary.main,
+    },
+  },
+})
+
+const TruncatedSuffix: FC<{
+  post: PostsList,
+  wordsLeft: number,
+}> = ({post, wordsLeft}) => {
+  return (
+    <Link to={postGetPageUrl(post)} eventProps={{intent: 'expandPost'}}>
+        {`(Continue Reading - ${wordsLeft} words more)`}
+    </Link>
+  );
+}
+
+const foreignFetchProps = {
+  collectionName: "Posts",
+  fragmentName: "PostsList",
+} as const;
+
+const expandedFetchProps = {
+  collectionName: "Posts",
+  fragmentName: "PostsExpandedHighlight",
+  fetchPolicy: "cache-first",
+} as const;
+
+const FeedPostHighlightBody = ({
+  post,
+  maxCollapsedLengthWords,
+  expanded,
+  setExpanded,
+  expandedLoading,
+  expandedDocument,
+  classes,
+}: {
+  post: PostsList,
+  maxCollapsedLengthWords: number,
+  expanded: boolean,
+  setExpanded: (value: boolean) => void,
+  expandedLoading: boolean,
+  expandedDocument?: PostsExpandedHighlight,
+  classes: ClassesType,
+}) => {
+  const { htmlHighlight = "", wordCount = 0 } = post.contents || {};
+
+  const { recordPostView } = useRecordPostView(post); 
+  const { captureEvent } = useTracking();
+
+  const clickExpand = useCallback((ev: MouseEvent) => {
+    setExpanded(true);
+    ev.preventDefault();
+    void recordPostView({post, extraEventProperties: {type: 'expandPostCard'}});
+    captureEvent('readMoreClicked', {postId: post._id});
+  }, [setExpanded, recordPostView, post, captureEvent]);
+
+
+  const maxLengthWords = expanded ? 1000 : maxCollapsedLengthWords;
+
+  const html = expandedDocument?.contents?.html || htmlHighlight
```

Comment:
prefer ?? (assuming the behavior we want when we have `html` which is an empty string is to use that, rather than coalescing)